### PR TITLE
#656 Hide content for non registered user

### PR DESCRIPTION
**Pull request by CarlosNZ**
Implements #656. Hides all page content except that which is relevant to user registration from "nonRegistered" user.

Use [back-end branch from PR #334](https://github.com/openmsupply/application-manager-server/pull/334) to see changes in UserRegistration template

As part of this, I've tidied up some of the logic handling the nonRegistered user to make sure they'll always be logged out when trying to access something other than user registration template. Also tweaked a little bit where some "placeholder" content was showing up before certain things loaded, which looked a little messy (those components now return `null` if certain things aren't loaded yet).

### Additional tweaks:

- added a "devCSS.css" file -- uncomment the "div" definition to see all divs on page outlined
- "TextInfo" element parses "title" parameter as Markdown

### CODE REVIEW

**Review by andreievg — COMMENTED** [review]
A couple of comment (this was easier then creating an issue).

Also as per our convo, 'isNonRegistered' should be returned from UserProvider;

I think it would simplify things a bit if server returns nonRegistered user (always), if JWT is undefined or is incorrect. That was kind of the intent form the get go.

Route to `application`, `login` are allowed for nonRegister user (route register is not needed). 

Something that is related, but not quite part of this feature (I think they can be done in the same PR)

userCreateApplication should be aligned now with other mutation hooks (they just return an async method)

ApplicationCreate page simplified to use new hook.

Then 'register' button, instead of navigating to create page,

File: src/components/Application/ApplicationContainer.tsx [hunk 1]
```diff
@@ -15,9 +15,10 @@ const ApplicationContainer: React.FC<ApplicationContainerProps> = ({
 }) => {
   const { replace } = useRouter()
   const { code, name } = template
+  const isNonRegistered = currentUser?.username === strings.USER_NONREGISTERED
   return (
-    <Container id="application-area">
-      <div className="top-container">
+    <Container id="application-area" className={isNonRegistered ? 'non-registered' : ''}>
+      <div className={'top-container' + isNonRegistered ? 'hidden-element' : ''}>
```
[the remark this answers is not in the document]
Comment (CarlosNZ) [reply src/components/Application/ApplicationContainer.tsx:21]: They both do slightly different things:
- `.non-registered` class shifts the whole container *up* now that the sticky UserArea is hidden
- `.hidden-element` hides the "top-container" (which has orgName and back label)

So the result is: we see "application-area", but without the top container component and all shifted up the page.

File: src/containers/User/UserArea.tsx [hunk 2]
```diff
@@ -12,6 +12,8 @@ const UserArea: React.FC = () => {
     userState: { currentUser },
   } = useUserState()
 
+  if (!currentUser || currentUser?.username === strings.USER_NONREGISTERED) return null
```
[the remark this answers is not in the document]
Comment (CarlosNZ) [reply src/containers/User/UserArea.tsx:15]: I thought we discussed this yesterday and arrived at hiding the whole UserArea is better. There's nothing of use in the UserArea for a non-registered user, so why show anything?

**Comment by CarlosNZ** [comment]
> Something that is related, but not quite part of this feature (I think they can be done in the same PR)
> 
> userCreateApplication should be aligned now with other mutation hooks (they just return an async method)
> 
> ApplicationCreate page simplified to use new hook.

Not quite sure what's needed here. There's an `onCompleted` method that is passed in by ApplicationCreate, not sure where we handle that.

FYI -- new working branch is `#659-nonRegistered-fixes` (currently has changes to return `isNonRegistered` from UserState

**Pull request merged**